x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -186,27 +225,45 @@ export class ESSearchSource extends AbstractESSource {
   }
 
   // Returns sort content for an Elasticsearch search body
-  _buildEsSort() {
+  _buildEsSort(): Array<Record<string, SortDirection | SortDirectionNumeric>> {
     const { sortField, sortOrder } = this._descriptor;
+
+    if (!sortField) {
+      throw new Error('Cannot build sort');
+    }
     return [
       {
         [sortField]: {
-          order: sortOrder,
+          order: (sortOrder as unknown) as SortDirection,
         },
       },
     ];
   }
 
-  async _getTopHits(layerName, searchFilters, registerCancelCallback) {
+  async _getTopHits(
+    layerName: string,
+    searchFilters: VectorSourceRequestMeta,
+    registerCancelCallback: (callback: () => void) => void
+  ) {
     const { topHitsSplitField: topHitsSplitFieldName, topHitsSize } = this._descriptor;
 
-    const indexPattern = await this.getIndexPattern();
+    if (!topHitsSplitFieldName) {
+      throw new Error('Cannot _getTopHits without topHitsSplitField');
+    }
+
+    const indexPattern: IndexPattern = await this.getIndexPattern();
 
     const { docValueFields, sourceOnlyFields, scriptFields } = getDocValueAndSourceFields(
       indexPattern,
       searchFilters.fieldNames
     );
-    const topHits = {
+    const topHits: {
+      size: number | undefined;
```

Comment:
This will always be `number`. `descriptor.topHitsSize` is defaulted to 1 in the constructor so even if its not provided it will be set in this method.